R/agg_scale.R

```diff
@@ -57,6 +55,53 @@
 #' @return \[`data.table()`\] with `id_cols` and `value_cols` columns for
 #'   requested aggregates or with scaled values.
 #'
+#' @section Severity Arguments:
+#' **`missing_dt_severity`**:
+#'
+#' Check for missing levels of `col_stem`, the variable being aggregated or
+#' scaled over.
+#' 1. `stop`: throw error (this is the default).
+#' 2. `warning` or `message`: throw warning/message and continue with
+#' aggregation/scaling where possible.
+#' 3. `none`: don't throw error or warning, continue with aggregation/scaling
+#' where possible.
+#' 4. `skip`: skip this check and continue with aggregation/scaling.
+#'
+#' **`present_agg_severity`** (`agg` only):
+#'
+#' Check for requested aggregates in `mapping` that are already present
+#' 1. `stop`: throw error (this is the default).
+#' 2. `warning` or `message`: throw warning/message, drop aggregates and continue
+#' with aggregation.
+#' 3. `none`: don't throw error or warning, drop aggregates and continue with
+#' aggregation.
+#' 4. `skip`: skip this check and add to the values already present for the
+#' aggregates.
+#'
+#' **`na_value_severity`**:
+#'
+#' Check for 'NA' values in the `value_cols`.
+#' 1. `stop`: throw error (this is the default).
+#' 2. `warning` or `message`: throw warning/message, drop missing values and
+#' continue with aggregation/scaling where possible (this likely will cause
+#' another error because of `missing_dt_severity`.
+#' 3. `none`: don't throw error or warning, drop missing values and continue
+#' with aggregation/scaling where possible (this likely will cause another error
+#' because of `missing_dt_severity`.
+#' 4. `skip`: skip this check and propagate `NA` values through
+#' aggregation/scaling.
+#'
+#' **`overlapping_dt_severity`**:
+#' Check for overlapping intervals that prevent collapsing to the most detailed
+#' common set of intervals. Or check for overlapping intervals in `col_stem`
+#' when aggregating/scaling.
+#' 1. `stop`: throw error (this is the default).
+#' 2. `warning` or `message`: throw warning/message, drop overlapping intervals
+#' and continue with aggregation/scaling.
+#' 3 `none`: don't throw error or warning, drop overlapping intervals and
```

Comment:
If you drop overlapping intervals you'd probably also need to specify something for `missing_dt_severity`, right?

Reply:
Updated to
```
#' 2. `warning` or `message`: throw warning/message, drop overlapping intervals
#' and continue with aggregation/scaling where possible (this may cause another
#' error because of `missing_dt_severity`).
#' 3 `none`: don't throw error or warning, drop overlapping intervals and
#' continue with aggregation/scaling where possible (this may cause another
#' error because of `missing_dt_severity`).
```